extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/AbstractMongoClientProducer.java

```diff
@@ -0,0 +1,296 @@
+package io.quarkus.mongodb.runtime;
+
+import static com.mongodb.AuthenticationMechanism.*;
+
+import java.lang.reflect.Constructor;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.annotation.PreDestroy;
+
+import org.bson.codecs.configuration.CodecProvider;
+import org.bson.codecs.configuration.CodecRegistries;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.Conventions;
+import org.bson.codecs.pojo.PojoCodecProvider;
+import org.jboss.logging.Logger;
+
+import com.mongodb.AuthenticationMechanism;
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCredential;
+import com.mongodb.MongoException;
+import com.mongodb.ReadPreference;
+import com.mongodb.ServerAddress;
+import com.mongodb.WriteConcern;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.connection.ClusterConnectionMode;
+
+import io.quarkus.mongodb.ReactiveMongoClient;
+import io.quarkus.mongodb.impl.ReactiveMongoClientImpl;
+
+public abstract class AbstractMongoClientProducer {
+
+    private static final Logger LOGGER = Logger.getLogger(AbstractMongoClientProducer.class.getName());
+    private static final Pattern COLON_PATTERN = Pattern.compile(":");
+    private MongodbConfig config;
+    //private boolean disableSslSupport = false;
+    private List<String> codecProviders;
+    private List<MongoClient> mongoclients = new ArrayList<>();
+    private List<ReactiveMongoClient> reactiveMongoClients = new ArrayList<>();
+
+    @SuppressWarnings("unused")
```

Comment:
it was because I get a format error during build at start and I think it was because of warning on idea about unused function. I can removed it if you want.